assimilation_code/modules/utilities/netcdf_utilities_mod.f90

```diff
@@ -437,7 +485,7 @@ subroutine nc_add_char_att_to_var(ncid, varname, attname, val, context, filename
 ret = nf90_inq_varid(ncid, varname, varid)
 call nc_check(ret, routine, 'inquire variable id for '//trim(varname), context, filename, ncid)
 
-ret = nf90_put_att(ncid, varid, attname, val)
+ret = nf90_put_att(ncid, varid, attname, adjustl(val))
```

Comment:
I was surprised by this, but it seemed to be needed.

Reply:
in your calling code how were you creating the val string that you passed in here?  if using write() without a format string it might have added an initial blank?  i'm not sure if this is a good idea to muck with what the user gives us here.  (e.g. what if they want 8 initial blanks for some reason?)

Reply:
I removed the adjustl() and nothing broke ... I think the code I was using that exercised that is no longer relevant.